builtin/providers/github/resource_github_repository.go

```diff
@@ -0,0 +1,165 @@
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"homepage_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"private": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"has_issues": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"has_wiki": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"has_downloads": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"auto_init": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"full_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"default_branch": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ssh_clone_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"svn_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"git_clone_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"http_clone_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGithubRepositoryObject(d *schema.ResourceData) *github.Repository {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	homepageUrl := d.Get("homepage_url").(string)
+	private := d.Get("private").(bool)
+	hasIssues := d.Get("has_issues").(bool)
+	hasWiki := d.Get("has_wiki").(bool)
+	hasDownloads := d.Get("has_downloads").(bool)
+	autoInit := d.Get("auto_init").(bool)
+
+	repo := &github.Repository{
+		Name:         &name,
+		Description:  &description,
+		Homepage:     &homepageUrl,
+		Private:      &private,
+		HasIssues:    &hasIssues,
+		HasWiki:      &hasWiki,
+		HasDownloads: &hasDownloads,
+		AutoInit:     &autoInit,
+	}
+
+	return repo
+}
+
+func resourceGithubRepositoryCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+
+	repoReq := resourceGithubRepositoryObject(d)
+	log.Printf("[DEBUG] create github repository %s/%s", meta.(*Organization).name, *repoReq.Name)
+	repo, _, err := client.Repositories.Create(meta.(*Organization).name, repoReq)
+	if err != nil {
+		return err
+	}
+	d.SetId(*repo.Name)
+
+	return resourceGithubRepositoryRead(d, meta)
+}
+
+func resourceGithubRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+	repoName := d.Id()
+
+	log.Printf("[DEBUG] read github repository %s/%s", meta.(*Organization).name, repoName)
+	repo, resp, err := client.Repositories.Get(meta.(*Organization).name, repoName)
+	if err != nil {
+		if resp.StatusCode == 404 {
+			d.SetId("")
```

Comment:
nitpick: when this happens, let's log a `[DEBUG]` that says what/why we've done here

```
log.Printf("[DEBUG] removing (%s) from state because reasons", d.Id())
d.SetId("")
```